sql/catalyst/src/main/scala/org/apache/spark/sql/types/DecimalType.scala

```diff
@@ -140,7 +140,12 @@ object DecimalType extends AbstractDataType {
   }
 
   private[sql] def bounded(precision: Int, scale: Int): DecimalType = {
-    DecimalType(min(precision, MAX_PRECISION), min(scale, MAX_SCALE))
```

Comment:
I will take your word for the risk involved, I am very new to this project.

From a layman's perspective, it seems that doing some basic checks when instantiating the type would make the type more robust. If I understand correctly a `precision <= 0` is not allowed, so this patch returns a /default/ decimal. Similarly, a `scale > precision` is not allowed, so it returns a decimal with the scale truncated to the size of the precision. My thoughts are that this will catch unexpected inputs and still behave in an expected way. Users instantiating these decimals in ways are intended will still get the same type back.